addons/l10n_it_edi/models/ir_mail_server.py

```diff
@@ -123,35 +146,48 @@ def _attachment_invoice(self, msg_txt):
                     _logger.info('New E-invoice in zip file: %s', attachment.fname)
                     self._create_invoice_from_mail_with_zip(attachment, from_address)
 
-    def _create_invoice_from_mail(self, att_content, att_name, from_address):
-        if self.env['account.move'].search([('l10n_it_einvoice_name', '=', att_name)], limit=1):
-            # invoice already exist
+    def _create_invoice_from_mail(self, att_content_data, att_name, from_address):
```

Comment:
any particular reason that `att_content` has been renamed to `att_content_data` other than the fact that it's the name of the variable when this method is called ? IMO, it would be better to keep the changes as minimal as possible.

Reply:
It's meant to bring awareness that the content is now bytes instead of string. 
The "fromstring" accepting bytes *and* strings is very confusing to me.

Reply:
Ok !

Reply:
Changing the parameter type is not a good idea in stable. Is it not possible to do the casting yourself?

Reply:
No, because this function was called once with bytes (in case of zip attachments) and once with strings (in case of single XML file attachment) already.  
So, it's not really a change, it's a clarification.
(In this case, it was less risky and complicated to translate a single XML file to bytes, than to make a string out of the zip content, and then having again to put it back as bytes for the etree to parse)

Reply:
Then it might be the good time to add some docstring

Reply:
Docstring added.
https://github.com/odoo-dev/odoo/commit/4af4d458ddd05ac211270cf7ee1305b5e423da91#diff-c3847f08e297c87760d4378c7748d0313abd2068c7fdaccb5be2d4e8a41ff692R149

```py
    def _create_invoice_from_mail(self, att_content_data, att_name, from_address):
        """ Creates an invoice from the content of an email present in ir.attachments

        :param att_content_data:   The 'utf-8' encoded bytes string representing the content of the attachment.
        :param att_name:           The attachment's file name.
        :param from_address:       The sender address of the email.
        """
```